code/datums/elements/decals/blood.dm

```diff
@@ -18,8 +18,8 @@
 		_icon = 'icons/effects/blood.dmi'
 	if(!_icon_state)
 		_icon_state = "itemblood"
-	var/icon = initial(I.icon)
```

Comment:
I assume it was initial for a reason, is there another way you could go about fixing this?

Reply:
Purely technically I could just fetch the greyscale icon and keep the initial. Should I do that?

Reply:
I'd like you to try and figure out why this uses initial, and then if no answer surfaces we could add a flag to atoms or whatever that tells this proc to use their actual icon rather then initial.

it's a hack, and I'd like it to be not needed and the current implementation to be the proper solution, but we need to know why things are the way they are first

Reply:
Initial is used here she to icon caching really, that's why.
I'll take a look a bit later today again, but during my testing so far it seemed to be okay without it.

Reply:
Frankly I think this was initial "just in case".

I made this test with cowboy shoes (non-GAGS) and black sneakers (GAGS):
1. Cover shoes in blood, check ref
2. Clean shoes, check ref
3. Make shoes damaged by setting self on fire
4. Try making damaged shoes dirty with blood, check ref

The ref did not change, and it caches fine.
![image](https://user-images.githubusercontent.com/81999976/116616283-36133c80-a945-11eb-9889-f5fd4e4849d6.png)
Despite them having the same ref in the function though, both in the burned and not burned state, they seem to be working fine and correctly.

Good to merge, I believe.

Reply:
What a helpful commit desc https://github.com/tgstation/tgstation/commit/cff9ed5f7103fcb69a6bb67c2894b681bcf6cd28